homeassistant/components/lock/nuki.py

```diff
@@ -0,0 +1,72 @@
+"""
+Nuki.io lock platform.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/lock.nuki/
+"""
+from datetime import timedelta
+import logging
+import voluptuous as vol
+
+from homeassistant.components.lock import (LockDevice, PLATFORM_SCHEMA)
+from homeassistant.const import (CONF_HOST, CONF_PORT, CONF_TOKEN)
+from homeassistant.util import Throttle
+import homeassistant.helpers.config_validation as cv
+
+
+REQUIREMENTS = ['pynuki==1.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_PORT = 8080
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    vol.Required(CONF_TOKEN): cv.string
+})
+
+
+MIN_TIME_BETWEEN_SCANS = timedelta(seconds=30)
+MIN_TIME_BETWEEN_FORCED_SCANS = timedelta(seconds=5)
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Demo lock platform."""
+    from pynuki import NukiBridge
+    bridge = NukiBridge(config.get(CONF_HOST), config.get(CONF_TOKEN))
+    add_devices([NukiLock(lock) for lock in bridge.locks])
+
+
+class NukiLock(LockDevice):
+    """Representation of a Nuki lock."""
+
+    def __init__(self, nuki_lock):
+        """Initialize the lock."""
+        self._nuki_lock = nuki_lock
+
+    @property
+    def name(self):
+        """Return the name of the lock."""
+        return self._nuki_lock.name
+
+    @property
+    def is_locked(self):
+        """Return true if lock is locked."""
+        return self._nuki_lock.is_locked
+
+    @Throttle(MIN_TIME_BETWEEN_SCANS, MIN_TIME_BETWEEN_FORCED_SCANS)
+    def update(self):
+        """Update the nuki lock properties."""
+        self._nuki_lock.update()
+
+    def lock(self, **kwargs):
+        """Lock the device."""
+        self._nuki_lock.lock()
+        self.update_ha_state()
```

Comment:
Could you elaborate a little more? Without this `lock` method, how would HASS know which method to call on my Nuki Lock to lock it?